/* tslint:disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

/**
 * JSON schema for polymer entity core.
 */
export interface CorePolymerEntity {
  rcsb_cluster_membership?: {
    /**
     * Identifier for a cluster at the specified level of sequence identity within the cluster data set.
     */
    cluster_id?: number;
    /**
     * Sequence identity expressed as an integer percent value.
     */
    identity?: number;
  }[];
  entity_poly?: {
    /**
     * A flag to indicate whether the polymer contains at least
     *  one monomer-to-monomer link different from that implied by
     *  _entity_poly.type.
     */
    nstd_linkage?: "n" | "no" | "y" | "yes";
    /**
     * A flag to indicate whether the polymer contains at least
     *  one monomer that is not considered standard.
     */
    nstd_monomer?: "n" | "no" | "y" | "yes";
    /**
     * Sequence of protein or nucleic acid polymer in standard one-letter
     *                codes of amino acids or nucleotides. Non-standard amino
     *                acids/nucleotides are represented by their Chemical
     *                Component Dictionary (CCD) codes in
     *                parenthesis. Deoxynucleotides are represented by the
     *                specially-assigned 2-letter CCD codes in parenthesis,
     *                with 'D' prefix added to their ribonucleotide
     *                counterparts. For hybrid polymer, each residue is
     *                represented by the code of its individual type. A
     *                cyclic polymer is represented in linear sequence from
     *                the chosen start to end.
     *
     * A for Alanine or Adenosine-5'-monophosphate
     * C for Cysteine or Cytidine-5'-monophosphate
     * D for Aspartic acid
     * E for Glutamic acid
     * F for Phenylalanine
     * G for Glycine or Guanosine-5'-monophosphate
     * H for Histidine
     * I for Isoleucine or Inosinic Acid
     * L for Leucine
     * K for Lysine
     * M for Methionine
     * N for Asparagine  or Unknown ribonucleotide
     * O for Pyrrolysine
     * P for Proline
     * Q for Glutamine
     * R for Arginine
     * S for Serine
     * T for Threonine
     * U for Selenocysteine or Uridine-5'-monophosphate
     * V for Valine
     * W for Tryptophan
     * Y for Tyrosine
     * (DA) for 2'-deoxyadenosine-5'-monophosphate
     * (DC) for 2'-deoxycytidine-5'-monophosphate
     * (DG) for 2'-deoxyguanosine-5'-monophosphate
     * (DT) for Thymidine-5'-monophosphate
     * (MSE) for Selenomethionine
     * (SEP) for Phosphoserine
     * (PTO) for Phosphothreonine
     * (PTR) for Phosphotyrosine
     * (PCA) for Pyroglutamic acid
     * (UNK) for Unknown amino acid
     * (ACE) for Acetylation cap
     * (NH2) for Amidation cap
     */
    pdbx_seq_one_letter_code?: string;
    /**
     * Canonical sequence of protein or nucleic acid polymer in standard
     *                one-letter codes of amino acids or nucleotides,
     *                corresponding to the sequence in
     *                _entity_poly.pdbx_seq_one_letter_code. Non-standard
     *                amino acids/nucleotides are represented by the codes of
     *                their parents if parent is specified in
     *                _chem_comp.mon_nstd_parent_comp_id, or by letter 'X' if
     *                parent is not specified. Deoxynucleotides are
     *                represented by their canonical one-letter codes of A,
     *                C, G, or T.
     *
     *                For modifications with several parent amino acids,
     * 	       all corresponding parent amino acid codes will be listed
     * 	       (ex. chromophores).
     */
    pdbx_seq_one_letter_code_can?: string;
    /**
     * Evidence for the assignment of the polymer sequence.
     */
    pdbx_sequence_evidence_code?: "depositor provided" | "derived from coordinates";
    /**
     * The PDB strand/chain id(s) corresponding to this polymer entity.
     */
    pdbx_strand_id?: string;
    /**
     * For Structural Genomics entries, the sequence's target identifier registered at the TargetTrack database.
     */
    pdbx_target_identifier?: string;
    /**
     * Number of regions in the sample sequence identified as expression tags, linkers, or
     *  cloning artifacts.
     */
    rcsb_artifact_monomer_count?: number;
    /**
     * Number of monomer conflicts relative to the reference sequence.
     */
    rcsb_conflict_count?: number;
    /**
     * Number of monomer deletions relative to the reference sequence.
     */
    rcsb_deletion_count?: number;
    /**
     * A coarse-grained polymer entity type.
     */
    rcsb_entity_polymer_type?: "DNA" | "NA-hybrid" | "Other" | "Protein" | "RNA";
    /**
     * Number of monomer insertions relative to the reference sequence.
     */
    rcsb_insertion_count?: number;
    /**
     * Number of engineered mutations engineered in the sample sequence.
     */
    rcsb_mutation_count?: number;
    /**
     * Number of non-standard monomers in the sample sequence.
     */
    rcsb_non_std_monomer_count?: number;
    rcsb_non_std_monomers?: string[];
    /**
     * For polymer BIRD molecules the BIRD identifier for the entity.
     */
    rcsb_prd_id?: string;
    /**
     * The monomer length of the sample sequence.
     */
    rcsb_sample_sequence_length?: number;
    /**
     * The type of the polymer.
     */
    type?:
      | "cyclic-pseudo-peptide"
      | "other"
      | "peptide nucleic acid"
      | "polydeoxyribonucleotide"
      | "polydeoxyribonucleotide/polyribonucleotide hybrid"
      | "polypeptide(D)"
      | "polypeptide(L)"
      | "polyribonucleotide";
  };
  /**
   * @minItems 1
   */
  entity_src_gen?: [
    {
      /**
       * A unique identifier for the expression system. This
       *  should be extracted from a local list of expression
       *  systems.
       */
      expression_system_id?: string;
      /**
       * The common name of the natural organism from which the gene was
       *  obtained.
       */
      gene_src_common_name?: string;
      /**
       * A description of special aspects of the natural organism from
       *  which the gene was obtained.
       */
      gene_src_details?: string;
      /**
       * The genus of the natural organism from which the gene was
       *  obtained.
       */
      gene_src_genus?: string;
      /**
       * The species of the natural organism from which the gene was
       *  obtained.
       */
      gene_src_species?: string;
      /**
       * The strain of the natural organism from which the gene was
       *  obtained, if relevant.
       */
      gene_src_strain?: string;
      /**
       * The tissue of the natural organism from which the gene was
       *  obtained.
       */
      gene_src_tissue?: string;
      /**
       * The subcellular fraction of the tissue of the natural organism
       *  from which the gene was obtained.
       */
      gene_src_tissue_fraction?: string;
      /**
       * The common name of the organism that served as host for the
       *  production of the entity.  Where full details of the protein
       *  production are available it would be expected that this item
       *  be derived from _entity_src_gen_express.host_org_common_name
       *  or via _entity_src_gen_express.host_org_tax_id
       */
      host_org_common_name?: string;
      /**
       * A description of special aspects of the organism that served as
       *  host for the production of the entity. Where full details of
       *  the protein production are available it would be expected that
       *  this item would derived from _entity_src_gen_express.host_org_details
       */
      host_org_details?: string;
      /**
       * The genus of the organism that served as host for the production
       *  of the entity.
       */
      host_org_genus?: string;
      /**
       * The species of the organism that served as host for the
       *  production of the entity.
       */
      host_org_species?: string;
      /**
       * This data item identifies cases in which an alternative source
       *  modeled.
       */
      pdbx_alt_source_flag?: "model" | "sample";
      /**
       * The beginning polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_beg_seq_num?: number;
      /**
       * Information on the source which is not given elsewhere.
       */
      pdbx_description?: string;
      /**
       * The ending polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_end_seq_num?: number;
      /**
       * American Type Culture Collection tissue culture number.
       */
      pdbx_gene_src_atcc?: string;
      /**
       * Cell type.
       */
      pdbx_gene_src_cell?: string;
      /**
       * The specific line of cells.
       */
      pdbx_gene_src_cell_line?: string;
      /**
       * Identifies the location inside (or outside) the cell.
       */
      pdbx_gene_src_cellular_location?: string;
      /**
       * A domain or fragment of the molecule.
       */
      pdbx_gene_src_fragment?: string;
      /**
       * Identifies the gene.
       */
      pdbx_gene_src_gene?: string;
      /**
       * NCBI Taxonomy identifier for the gene source organism.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      pdbx_gene_src_ncbi_taxonomy_id?: string;
      /**
       * Organized group of tissues that carries on a specialized function.
       */
      pdbx_gene_src_organ?: string;
      /**
       * Organized structure within cell.
       */
      pdbx_gene_src_organelle?: string;
      /**
       * Scientific name of the organism.
       */
      pdbx_gene_src_scientific_name?: string;
      /**
       * Identifies the variant.
       */
      pdbx_gene_src_variant?: string;
      /**
       * Americal Tissue Culture Collection of the expression system. Where
       *  full details of the protein production are available it would
       *  be expected that this item  would be derived from
       *  _entity_src_gen_express.host_org_culture_collection
       */
      pdbx_host_org_atcc?: string;
      /**
       * Cell type from which the gene is derived. Where
       *  entity.target_id is provided this should be derived from
       *  details of the target.
       */
      pdbx_host_org_cell?: string;
      /**
       * A specific line of cells used as the expression system. Where
       *  full details of the protein production are available it would
       *  be expected that this item would be derived from
       *  entity_src_gen_express.host_org_cell_line
       */
      pdbx_host_org_cell_line?: string;
      /**
       * Identifies the location inside (or outside) the cell which
       *  expressed the molecule.
       */
      pdbx_host_org_cellular_location?: string;
      /**
       * Culture collection of the expression system. Where
       *  full details of the protein production are available it would
       *  be expected that this item  would be derived somehwere, but
       *  exactly where is not clear.
       */
      pdbx_host_org_culture_collection?: string;
      /**
       * Specific gene which expressed the molecule.
       */
      pdbx_host_org_gene?: string;
      /**
       * NCBI Taxonomy identifier for the expression system organism.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      pdbx_host_org_ncbi_taxonomy_id?: string;
      /**
       * Specific organ which expressed the molecule.
       */
      pdbx_host_org_organ?: string;
      /**
       * Specific organelle which expressed the molecule.
       */
      pdbx_host_org_organelle?: string;
      /**
       * The scientific name of the organism that served as host for the
       *  production of the entity. Where full details of the protein
       *  production are available it would be expected that this item
       *  would be derived from _entity_src_gen_express.host_org_scientific_name
       *  or via _entity_src_gen_express.host_org_tax_id
       */
      pdbx_host_org_scientific_name?: string;
      /**
       * The strain of the organism in which the entity was
       * expressed.
       */
      pdbx_host_org_strain?: string;
      /**
       * The specific tissue which expressed the molecule. Where full details
       *  of the protein production are available it would be expected that this
       *  item would be derived from _entity_src_gen_express.host_org_tissue
       */
      pdbx_host_org_tissue?: string;
      /**
       * The fraction of the tissue which expressed the
       * molecule.
       */
      pdbx_host_org_tissue_fraction?: string;
      /**
       * Variant of the organism used as the expression system. Where
       *  full details of the protein production are available it would
       *  be expected that this item be derived from
       *  entity_src_gen_express.host_org_variant or via
       *  _entity_src_gen_express.host_org_tax_id
       */
      pdbx_host_org_variant?: string;
      /**
       * Identifies the vector used. Where full details of the protein
       *  production are available it would be expected that this item
       *  would be derived from _entity_src_gen_clone.vector_name.
       */
      pdbx_host_org_vector?: string;
      /**
       * Identifies the type of vector used (plasmid, virus, or cosmid).
       *  Where full details of the protein production are available it
       *  would be expected that this item would be derived from
       *  _entity_src_gen_express.vector_type.
       */
      pdbx_host_org_vector_type?: string;
      /**
       * This data item povides additional information about the sequence type.
       */
      pdbx_seq_type?: "Biological sequence" | "C-terminal tag" | "Linker" | "N-terminal tag";
      /**
       * This data item is an ordinal identifier for entity_src_gen data records.
       */
      pdbx_src_id: number;
      /**
       * A description of special aspects of the plasmid that produced the
       *  entity in the host organism. Where full details of the protein
       *  production are available it would be expected that this item
       *  would be derived from _pdbx_construct.details of the construct
       *  pointed to from _entity_src_gen_express.plasmid_id.
       */
      plasmid_details?: string;
      /**
       * The name of the plasmid that produced the entity in the host
       *  organism. Where full details of the protein production are available
       *  it would be expected that this item would be derived from
       *  _pdbx_construct.name of the construct pointed to from
       *  _entity_src_gen_express.plasmid_id.
       */
      plasmid_name?: string;
    },
    ...{
      /**
       * A unique identifier for the expression system. This
       *  should be extracted from a local list of expression
       *  systems.
       */
      expression_system_id?: string;
      /**
       * The common name of the natural organism from which the gene was
       *  obtained.
       */
      gene_src_common_name?: string;
      /**
       * A description of special aspects of the natural organism from
       *  which the gene was obtained.
       */
      gene_src_details?: string;
      /**
       * The genus of the natural organism from which the gene was
       *  obtained.
       */
      gene_src_genus?: string;
      /**
       * The species of the natural organism from which the gene was
       *  obtained.
       */
      gene_src_species?: string;
      /**
       * The strain of the natural organism from which the gene was
       *  obtained, if relevant.
       */
      gene_src_strain?: string;
      /**
       * The tissue of the natural organism from which the gene was
       *  obtained.
       */
      gene_src_tissue?: string;
      /**
       * The subcellular fraction of the tissue of the natural organism
       *  from which the gene was obtained.
       */
      gene_src_tissue_fraction?: string;
      /**
       * The common name of the organism that served as host for the
       *  production of the entity.  Where full details of the protein
       *  production are available it would be expected that this item
       *  be derived from _entity_src_gen_express.host_org_common_name
       *  or via _entity_src_gen_express.host_org_tax_id
       */
      host_org_common_name?: string;
      /**
       * A description of special aspects of the organism that served as
       *  host for the production of the entity. Where full details of
       *  the protein production are available it would be expected that
       *  this item would derived from _entity_src_gen_express.host_org_details
       */
      host_org_details?: string;
      /**
       * The genus of the organism that served as host for the production
       *  of the entity.
       */
      host_org_genus?: string;
      /**
       * The species of the organism that served as host for the
       *  production of the entity.
       */
      host_org_species?: string;
      /**
       * This data item identifies cases in which an alternative source
       *  modeled.
       */
      pdbx_alt_source_flag?: "model" | "sample";
      /**
       * The beginning polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_beg_seq_num?: number;
      /**
       * Information on the source which is not given elsewhere.
       */
      pdbx_description?: string;
      /**
       * The ending polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_end_seq_num?: number;
      /**
       * American Type Culture Collection tissue culture number.
       */
      pdbx_gene_src_atcc?: string;
      /**
       * Cell type.
       */
      pdbx_gene_src_cell?: string;
      /**
       * The specific line of cells.
       */
      pdbx_gene_src_cell_line?: string;
      /**
       * Identifies the location inside (or outside) the cell.
       */
      pdbx_gene_src_cellular_location?: string;
      /**
       * A domain or fragment of the molecule.
       */
      pdbx_gene_src_fragment?: string;
      /**
       * Identifies the gene.
       */
      pdbx_gene_src_gene?: string;
      /**
       * NCBI Taxonomy identifier for the gene source organism.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      pdbx_gene_src_ncbi_taxonomy_id?: string;
      /**
       * Organized group of tissues that carries on a specialized function.
       */
      pdbx_gene_src_organ?: string;
      /**
       * Organized structure within cell.
       */
      pdbx_gene_src_organelle?: string;
      /**
       * Scientific name of the organism.
       */
      pdbx_gene_src_scientific_name?: string;
      /**
       * Identifies the variant.
       */
      pdbx_gene_src_variant?: string;
      /**
       * Americal Tissue Culture Collection of the expression system. Where
       *  full details of the protein production are available it would
       *  be expected that this item  would be derived from
       *  _entity_src_gen_express.host_org_culture_collection
       */
      pdbx_host_org_atcc?: string;
      /**
       * Cell type from which the gene is derived. Where
       *  entity.target_id is provided this should be derived from
       *  details of the target.
       */
      pdbx_host_org_cell?: string;
      /**
       * A specific line of cells used as the expression system. Where
       *  full details of the protein production are available it would
       *  be expected that this item would be derived from
       *  entity_src_gen_express.host_org_cell_line
       */
      pdbx_host_org_cell_line?: string;
      /**
       * Identifies the location inside (or outside) the cell which
       *  expressed the molecule.
       */
      pdbx_host_org_cellular_location?: string;
      /**
       * Culture collection of the expression system. Where
       *  full details of the protein production are available it would
       *  be expected that this item  would be derived somehwere, but
       *  exactly where is not clear.
       */
      pdbx_host_org_culture_collection?: string;
      /**
       * Specific gene which expressed the molecule.
       */
      pdbx_host_org_gene?: string;
      /**
       * NCBI Taxonomy identifier for the expression system organism.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      pdbx_host_org_ncbi_taxonomy_id?: string;
      /**
       * Specific organ which expressed the molecule.
       */
      pdbx_host_org_organ?: string;
      /**
       * Specific organelle which expressed the molecule.
       */
      pdbx_host_org_organelle?: string;
      /**
       * The scientific name of the organism that served as host for the
       *  production of the entity. Where full details of the protein
       *  production are available it would be expected that this item
       *  would be derived from _entity_src_gen_express.host_org_scientific_name
       *  or via _entity_src_gen_express.host_org_tax_id
       */
      pdbx_host_org_scientific_name?: string;
      /**
       * The strain of the organism in which the entity was
       * expressed.
       */
      pdbx_host_org_strain?: string;
      /**
       * The specific tissue which expressed the molecule. Where full details
       *  of the protein production are available it would be expected that this
       *  item would be derived from _entity_src_gen_express.host_org_tissue
       */
      pdbx_host_org_tissue?: string;
      /**
       * The fraction of the tissue which expressed the
       * molecule.
       */
      pdbx_host_org_tissue_fraction?: string;
      /**
       * Variant of the organism used as the expression system. Where
       *  full details of the protein production are available it would
       *  be expected that this item be derived from
       *  entity_src_gen_express.host_org_variant or via
       *  _entity_src_gen_express.host_org_tax_id
       */
      pdbx_host_org_variant?: string;
      /**
       * Identifies the vector used. Where full details of the protein
       *  production are available it would be expected that this item
       *  would be derived from _entity_src_gen_clone.vector_name.
       */
      pdbx_host_org_vector?: string;
      /**
       * Identifies the type of vector used (plasmid, virus, or cosmid).
       *  Where full details of the protein production are available it
       *  would be expected that this item would be derived from
       *  _entity_src_gen_express.vector_type.
       */
      pdbx_host_org_vector_type?: string;
      /**
       * This data item povides additional information about the sequence type.
       */
      pdbx_seq_type?: "Biological sequence" | "C-terminal tag" | "Linker" | "N-terminal tag";
      /**
       * This data item is an ordinal identifier for entity_src_gen data records.
       */
      pdbx_src_id: number;
      /**
       * A description of special aspects of the plasmid that produced the
       *  entity in the host organism. Where full details of the protein
       *  production are available it would be expected that this item
       *  would be derived from _pdbx_construct.details of the construct
       *  pointed to from _entity_src_gen_express.plasmid_id.
       */
      plasmid_details?: string;
      /**
       * The name of the plasmid that produced the entity in the host
       *  organism. Where full details of the protein production are available
       *  it would be expected that this item would be derived from
       *  _pdbx_construct.name of the construct pointed to from
       *  _entity_src_gen_express.plasmid_id.
       */
      plasmid_name?: string;
    }[]
  ];
  /**
   * @minItems 1
   */
  entity_src_nat?: [
    {
      /**
       * The common name of the organism from which the entity
       *  was isolated.
       */
      common_name?: string;
      /**
       * A description of special aspects of the organism from which the
       *  entity was isolated.
       */
      details?: string;
      /**
       * The genus of the organism from which the entity was isolated.
       */
      genus?: string;
      /**
       * This data item identifies cases in which an alternative source
       *  modeled.
       */
      pdbx_alt_source_flag?: "model" | "sample";
      /**
       * Americal Tissue Culture Collection number.
       */
      pdbx_atcc?: string;
      /**
       * The beginning polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_beg_seq_num?: number;
      /**
       * A particular cell type.
       */
      pdbx_cell?: string;
      /**
       * The specific line of cells.
       */
      pdbx_cell_line?: string;
      /**
       * Identifies the location inside (or outside) the cell.
       */
      pdbx_cellular_location?: string;
      /**
       * The ending polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_end_seq_num?: number;
      /**
       * A domain or fragment of the molecule.
       */
      pdbx_fragment?: string;
      /**
       * NCBI Taxonomy identifier for the source organism.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      pdbx_ncbi_taxonomy_id?: string;
      /**
       * Organized group of tissues that carries on a specialized function.
       */
      pdbx_organ?: string;
      /**
       * Organized structure within cell.
       */
      pdbx_organelle?: string;
      /**
       * Scientific name of the organism of the natural source.
       */
      pdbx_organism_scientific?: string;
      /**
       * Details about the plasmid.
       */
      pdbx_plasmid_details?: string;
      /**
       * The plasmid containing the gene.
       */
      pdbx_plasmid_name?: string;
      /**
       * Identifies the secretion from which the molecule was isolated.
       */
      pdbx_secretion?: string;
      /**
       * This data item is an ordinal identifier for entity_src_nat data records.
       */
      pdbx_src_id: number;
      /**
       * Identifies the variant.
       */
      pdbx_variant?: string;
      /**
       * The species of the organism from which the entity was isolated.
       */
      species?: string;
      /**
       * The strain of the organism from which the entity was isolated.
       */
      strain?: string;
      /**
       * The tissue of the organism from which the entity was isolated.
       */
      tissue?: string;
      /**
       * The subcellular fraction of the tissue of the organism from
       *  which the entity was isolated.
       */
      tissue_fraction?: string;
    },
    ...{
      /**
       * The common name of the organism from which the entity
       *  was isolated.
       */
      common_name?: string;
      /**
       * A description of special aspects of the organism from which the
       *  entity was isolated.
       */
      details?: string;
      /**
       * The genus of the organism from which the entity was isolated.
       */
      genus?: string;
      /**
       * This data item identifies cases in which an alternative source
       *  modeled.
       */
      pdbx_alt_source_flag?: "model" | "sample";
      /**
       * Americal Tissue Culture Collection number.
       */
      pdbx_atcc?: string;
      /**
       * The beginning polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_beg_seq_num?: number;
      /**
       * A particular cell type.
       */
      pdbx_cell?: string;
      /**
       * The specific line of cells.
       */
      pdbx_cell_line?: string;
      /**
       * Identifies the location inside (or outside) the cell.
       */
      pdbx_cellular_location?: string;
      /**
       * The ending polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_end_seq_num?: number;
      /**
       * A domain or fragment of the molecule.
       */
      pdbx_fragment?: string;
      /**
       * NCBI Taxonomy identifier for the source organism.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      pdbx_ncbi_taxonomy_id?: string;
      /**
       * Organized group of tissues that carries on a specialized function.
       */
      pdbx_organ?: string;
      /**
       * Organized structure within cell.
       */
      pdbx_organelle?: string;
      /**
       * Scientific name of the organism of the natural source.
       */
      pdbx_organism_scientific?: string;
      /**
       * Details about the plasmid.
       */
      pdbx_plasmid_details?: string;
      /**
       * The plasmid containing the gene.
       */
      pdbx_plasmid_name?: string;
      /**
       * Identifies the secretion from which the molecule was isolated.
       */
      pdbx_secretion?: string;
      /**
       * This data item is an ordinal identifier for entity_src_nat data records.
       */
      pdbx_src_id: number;
      /**
       * Identifies the variant.
       */
      pdbx_variant?: string;
      /**
       * The species of the organism from which the entity was isolated.
       */
      species?: string;
      /**
       * The strain of the organism from which the entity was isolated.
       */
      strain?: string;
      /**
       * The tissue of the organism from which the entity was isolated.
       */
      tissue?: string;
      /**
       * The subcellular fraction of the tissue of the organism from
       *  which the entity was isolated.
       */
      tissue_fraction?: string;
    }[]
  ];
  /**
   * @minItems 1
   */
  pdbx_entity_src_syn?: [
    {
      /**
       * A description of special aspects of the source for the
       *  synthetic entity.
       */
      details?: string;
      /**
       * NCBI Taxonomy identifier of the organism from which the sequence of
       *  the synthetic entity was derived.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      ncbi_taxonomy_id?: string;
      /**
       * The common name of the organism from which the sequence of
       *  the synthetic entity was derived.
       */
      organism_common_name?: string;
      /**
       * The scientific name of the organism from which the sequence of
       *  the synthetic entity was derived.
       */
      organism_scientific?: string;
      /**
       * This data item identifies cases in which an alternative source
       *  modeled.
       */
      pdbx_alt_source_flag?: "model" | "sample";
      /**
       * The beginning polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_beg_seq_num?: number;
      /**
       * The ending polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_end_seq_num?: number;
      /**
       * This data item is an ordinal identifier for pdbx_entity_src_syn data records.
       */
      pdbx_src_id: number;
    },
    ...{
      /**
       * A description of special aspects of the source for the
       *  synthetic entity.
       */
      details?: string;
      /**
       * NCBI Taxonomy identifier of the organism from which the sequence of
       *  the synthetic entity was derived.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      ncbi_taxonomy_id?: string;
      /**
       * The common name of the organism from which the sequence of
       *  the synthetic entity was derived.
       */
      organism_common_name?: string;
      /**
       * The scientific name of the organism from which the sequence of
       *  the synthetic entity was derived.
       */
      organism_scientific?: string;
      /**
       * This data item identifies cases in which an alternative source
       *  modeled.
       */
      pdbx_alt_source_flag?: "model" | "sample";
      /**
       * The beginning polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_beg_seq_num?: number;
      /**
       * The ending polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      pdbx_end_seq_num?: number;
      /**
       * This data item is an ordinal identifier for pdbx_entity_src_syn data records.
       */
      pdbx_src_id: number;
    }[]
  ];
  /**
   * @minItems 1
   */
  rcsb_entity_host_organism?: [
    {
      /**
       * The beginning polymer sequence position for the polymer section corresponding
       *  to this host organism.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      beg_seq_num?: number;
      /**
       * The common name of the host organism
       */
      common_name?: string;
      /**
       * The ending polymer sequence position for the polymer section corresponding
       *  to this host organism.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      end_seq_num?: number;
      ncbi_common_names?: string[];
      /**
       * The parent scientific name in the NCBI taxonomy hierarchy (depth=1) associated with this taxonomy code.
       *
       * References:
       *
       * Sayers EW, Barrett T, Benson DA, Bryant SH, Canese K, Chetvernin V,
       * Church DM, DiCuccio M, Edgar R, Federhen S, Feolo M, Geer LY,
       * Helmberg W, Kapustin Y, Landsman D, Lipman DJ, Madden TL, Maglott DR,
       * Miller V, Mizrachi I, Ostell J, Pruitt KD, Schuler GD, Sequeira E,
       * Sherry ST, Shumway M, Sirotkin K, Souvorov A, Starchenko G,
       * Tatusova TA, Wagner L, Yaschenko E, Ye J (2009). Database resources
       * of the National Center for Biotechnology Information. Nucleic Acids
       * Res. 2009 Jan;37(Database issue):D5-15. Epub 2008 Oct 21.
       *
       * Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Sayers EW (2009).
       * GenBank. Nucleic Acids Res. 2009 Jan;37(Database issue):D26-31.
       * Epub 2008 Oct 21.
       */
      ncbi_parent_scientific_name?: string;
      /**
       * The scientific name associated with this taxonomy code aggregated by the NCBI Taxonomy Database.
       *
       *   This name corresponds to the taxonomy identifier assigned by the PDB depositor.
       *
       *
       * References:
       *
       * Sayers EW, Barrett T, Benson DA, Bryant SH, Canese K, Chetvernin V,
       * Church DM, DiCuccio M, Edgar R, Federhen S, Feolo M, Geer LY,
       * Helmberg W, Kapustin Y, Landsman D, Lipman DJ, Madden TL, Maglott DR,
       * Miller V, Mizrachi I, Ostell J, Pruitt KD, Schuler GD, Sequeira E,
       * Sherry ST, Shumway M, Sirotkin K, Souvorov A, Starchenko G,
       * Tatusova TA, Wagner L, Yaschenko E, Ye J (2009). Database resources
       * of the National Center for Biotechnology Information. Nucleic Acids
       * Res. 2009 Jan;37(Database issue):D5-15. Epub 2008 Oct 21.
       *
       * Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Sayers EW (2009).
       * GenBank. Nucleic Acids Res. 2009 Jan;37(Database issue):D26-31.
       * Epub 2008 Oct 21.
       */
      ncbi_scientific_name?: string;
      /**
       * NCBI Taxonomy identifier for the host organism.
       *
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      ncbi_taxonomy_id?: number;
      /**
       * An identifier for an entity segment.
       */
      pdbx_src_id: string;
      /**
       * A code indicating the provenance of the host organism.
       */
      provenance_source?: "PDB Primary Data" | "Primary Data";
      /**
       * The scientific name of the host organism
       */
      scientific_name?: string;
      taxonomy_lineage?: {
        /**
         * Members of the NCBI Taxonomy lineage as parent taxonomy lineage depth (1-N)
         */
        depth?: number;
        /**
         * Members of the NCBI Taxonomy lineage as parent taxonomy idcodes.
         */
        id?: string;
        /**
         * Members of the NCBI Taxonomy lineage as parent taxonomy names.
         */
        name?: string;
      }[];
    },
    ...{
      /**
       * The beginning polymer sequence position for the polymer section corresponding
       *  to this host organism.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      beg_seq_num?: number;
      /**
       * The common name of the host organism
       */
      common_name?: string;
      /**
       * The ending polymer sequence position for the polymer section corresponding
       *  to this host organism.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      end_seq_num?: number;
      ncbi_common_names?: string[];
      /**
       * The parent scientific name in the NCBI taxonomy hierarchy (depth=1) associated with this taxonomy code.
       *
       * References:
       *
       * Sayers EW, Barrett T, Benson DA, Bryant SH, Canese K, Chetvernin V,
       * Church DM, DiCuccio M, Edgar R, Federhen S, Feolo M, Geer LY,
       * Helmberg W, Kapustin Y, Landsman D, Lipman DJ, Madden TL, Maglott DR,
       * Miller V, Mizrachi I, Ostell J, Pruitt KD, Schuler GD, Sequeira E,
       * Sherry ST, Shumway M, Sirotkin K, Souvorov A, Starchenko G,
       * Tatusova TA, Wagner L, Yaschenko E, Ye J (2009). Database resources
       * of the National Center for Biotechnology Information. Nucleic Acids
       * Res. 2009 Jan;37(Database issue):D5-15. Epub 2008 Oct 21.
       *
       * Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Sayers EW (2009).
       * GenBank. Nucleic Acids Res. 2009 Jan;37(Database issue):D26-31.
       * Epub 2008 Oct 21.
       */
      ncbi_parent_scientific_name?: string;
      /**
       * The scientific name associated with this taxonomy code aggregated by the NCBI Taxonomy Database.
       *
       *   This name corresponds to the taxonomy identifier assigned by the PDB depositor.
       *
       *
       * References:
       *
       * Sayers EW, Barrett T, Benson DA, Bryant SH, Canese K, Chetvernin V,
       * Church DM, DiCuccio M, Edgar R, Federhen S, Feolo M, Geer LY,
       * Helmberg W, Kapustin Y, Landsman D, Lipman DJ, Madden TL, Maglott DR,
       * Miller V, Mizrachi I, Ostell J, Pruitt KD, Schuler GD, Sequeira E,
       * Sherry ST, Shumway M, Sirotkin K, Souvorov A, Starchenko G,
       * Tatusova TA, Wagner L, Yaschenko E, Ye J (2009). Database resources
       * of the National Center for Biotechnology Information. Nucleic Acids
       * Res. 2009 Jan;37(Database issue):D5-15. Epub 2008 Oct 21.
       *
       * Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Sayers EW (2009).
       * GenBank. Nucleic Acids Res. 2009 Jan;37(Database issue):D26-31.
       * Epub 2008 Oct 21.
       */
      ncbi_scientific_name?: string;
      /**
       * NCBI Taxonomy identifier for the host organism.
       *
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      ncbi_taxonomy_id?: number;
      /**
       * An identifier for an entity segment.
       */
      pdbx_src_id: string;
      /**
       * A code indicating the provenance of the host organism.
       */
      provenance_source?: "PDB Primary Data" | "Primary Data";
      /**
       * The scientific name of the host organism
       */
      scientific_name?: string;
      taxonomy_lineage?: {
        /**
         * Members of the NCBI Taxonomy lineage as parent taxonomy lineage depth (1-N)
         */
        depth?: number;
        /**
         * Members of the NCBI Taxonomy lineage as parent taxonomy idcodes.
         */
        id?: string;
        /**
         * Members of the NCBI Taxonomy lineage as parent taxonomy names.
         */
        name?: string;
      }[];
    }[]
  ];
  /**
   * @minItems 1
   */
  rcsb_entity_source_organism?: [
    {
      /**
       * The beginning polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      beg_seq_num?: number;
      /**
       * The common name for the source organism assigned by the PDB depositor.
       */
      common_name?: string;
      /**
       * The ending polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      end_seq_num?: number;
      ncbi_common_names?: string[];
      /**
       * A parent scientific name in the NCBI taxonomy hierarchy of the source organism assigned by the PDB depositor.
       *   For cellular organism this corresponds to a superkingdom (e.g., Archaea, Bacteria, Eukaryota).  For viruses this
       *   corresponds to a clade (e.g.  Adnaviria, Bicaudaviridae, Clavaviridae). For other and unclassified entries this
       *   corresponds to the first level of any taxonomic rank below the root level.
       *
       * References:
       *
       * Sayers EW, Barrett T, Benson DA, Bryant SH, Canese K, Chetvernin V,
       * Church DM, DiCuccio M, Edgar R, Federhen S, Feolo M, Geer LY,
       * Helmberg W, Kapustin Y, Landsman D, Lipman DJ, Madden TL, Maglott DR,
       * Miller V, Mizrachi I, Ostell J, Pruitt KD, Schuler GD, Sequeira E,
       * Sherry ST, Shumway M, Sirotkin K, Souvorov A, Starchenko G,
       * Tatusova TA, Wagner L, Yaschenko E, Ye J (2009). Database resources
       * of the National Center for Biotechnology Information. Nucleic Acids
       * Res. 2009 Jan;37(Database issue):D5-15. Epub 2008 Oct 21.
       *
       * Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Sayers EW (2009).
       * GenBank. Nucleic Acids Res. 2009 Jan;37(Database issue):D26-31.
       * Epub 2008 Oct 21.
       */
      ncbi_parent_scientific_name?: string;
      /**
       * The scientific name associated with this taxonomy code aggregated by the NCBI Taxonomy Database.
       *
       *   This name corresponds to the taxonomy identifier assigned by the PDB depositor.
       *
       *
       * References:
       *
       * Sayers EW, Barrett T, Benson DA, Bryant SH, Canese K, Chetvernin V,
       * Church DM, DiCuccio M, Edgar R, Federhen S, Feolo M, Geer LY,
       * Helmberg W, Kapustin Y, Landsman D, Lipman DJ, Madden TL, Maglott DR,
       * Miller V, Mizrachi I, Ostell J, Pruitt KD, Schuler GD, Sequeira E,
       * Sherry ST, Shumway M, Sirotkin K, Souvorov A, Starchenko G,
       * Tatusova TA, Wagner L, Yaschenko E, Ye J (2009). Database resources
       * of the National Center for Biotechnology Information. Nucleic Acids
       * Res. 2009 Jan;37(Database issue):D5-15. Epub 2008 Oct 21.
       *
       * Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Sayers EW (2009).
       * GenBank. Nucleic Acids Res. 2009 Jan;37(Database issue):D26-31.
       * Epub 2008 Oct 21.
       */
      ncbi_scientific_name?: string;
      /**
       * NCBI Taxonomy identifier for the gene source organism assigned by the PDB depositor.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      ncbi_taxonomy_id?: number;
      /**
       * An identifier for the entity segment.
       */
      pdbx_src_id: string;
      /**
       * Reference to the provenance of the source organism details for the entity.
       *  Primary data indicates information obtained from the same source as the structural model.
       *  UniProt and NCBI are provided as alternate sources of provenance for organism details.
       */
      provenance_source?: "NCBI" | "PDB Primary Data" | "Primary Data" | "UniProt";
      /**
       * The scientific name of the source organism assigned by the PDB depositor.
       */
      scientific_name?: string;
      /**
       * The source type for the entity
       */
      source_type?: "genetically engineered" | "natural" | "synthetic";
      taxonomy_lineage?: {
        /**
         * Members of the NCBI Taxonomy lineage as parent taxonomy lineage depth (1-N)
         */
        depth?: number;
        /**
         * Members of the NCBI Taxonomy lineage as parent taxonomy idcodes.
         */
        id?: string;
        /**
         * Memebers of the NCBI Taxonomy lineage as parent taxonomy names.
         */
        name?: string;
      }[];
      rcsb_gene_name?: {
        /**
         * A code indicating the provenance of the source organism details for the entity
         */
        provenance_source?: "NCBI" | "PDB Primary Data" | "Primary Data" | "UniProt";
        /**
         * Gene name.
         */
        value?: string;
      }[];
    },
    ...{
      /**
       * The beginning polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      beg_seq_num?: number;
      /**
       * The common name for the source organism assigned by the PDB depositor.
       */
      common_name?: string;
      /**
       * The ending polymer sequence position for the polymer section corresponding
       *  to this source.
       *
       *  A reference to the sequence position in the entity_poly category.
       */
      end_seq_num?: number;
      ncbi_common_names?: string[];
      /**
       * A parent scientific name in the NCBI taxonomy hierarchy of the source organism assigned by the PDB depositor.
       *   For cellular organism this corresponds to a superkingdom (e.g., Archaea, Bacteria, Eukaryota).  For viruses this
       *   corresponds to a clade (e.g.  Adnaviria, Bicaudaviridae, Clavaviridae). For other and unclassified entries this
       *   corresponds to the first level of any taxonomic rank below the root level.
       *
       * References:
       *
       * Sayers EW, Barrett T, Benson DA, Bryant SH, Canese K, Chetvernin V,
       * Church DM, DiCuccio M, Edgar R, Federhen S, Feolo M, Geer LY,
       * Helmberg W, Kapustin Y, Landsman D, Lipman DJ, Madden TL, Maglott DR,
       * Miller V, Mizrachi I, Ostell J, Pruitt KD, Schuler GD, Sequeira E,
       * Sherry ST, Shumway M, Sirotkin K, Souvorov A, Starchenko G,
       * Tatusova TA, Wagner L, Yaschenko E, Ye J (2009). Database resources
       * of the National Center for Biotechnology Information. Nucleic Acids
       * Res. 2009 Jan;37(Database issue):D5-15. Epub 2008 Oct 21.
       *
       * Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Sayers EW (2009).
       * GenBank. Nucleic Acids Res. 2009 Jan;37(Database issue):D26-31.
       * Epub 2008 Oct 21.
       */
      ncbi_parent_scientific_name?: string;
      /**
       * The scientific name associated with this taxonomy code aggregated by the NCBI Taxonomy Database.
       *
       *   This name corresponds to the taxonomy identifier assigned by the PDB depositor.
       *
       *
       * References:
       *
       * Sayers EW, Barrett T, Benson DA, Bryant SH, Canese K, Chetvernin V,
       * Church DM, DiCuccio M, Edgar R, Federhen S, Feolo M, Geer LY,
       * Helmberg W, Kapustin Y, Landsman D, Lipman DJ, Madden TL, Maglott DR,
       * Miller V, Mizrachi I, Ostell J, Pruitt KD, Schuler GD, Sequeira E,
       * Sherry ST, Shumway M, Sirotkin K, Souvorov A, Starchenko G,
       * Tatusova TA, Wagner L, Yaschenko E, Ye J (2009). Database resources
       * of the National Center for Biotechnology Information. Nucleic Acids
       * Res. 2009 Jan;37(Database issue):D5-15. Epub 2008 Oct 21.
       *
       * Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Sayers EW (2009).
       * GenBank. Nucleic Acids Res. 2009 Jan;37(Database issue):D26-31.
       * Epub 2008 Oct 21.
       */
      ncbi_scientific_name?: string;
      /**
       * NCBI Taxonomy identifier for the gene source organism assigned by the PDB depositor.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      ncbi_taxonomy_id?: number;
      /**
       * An identifier for the entity segment.
       */
      pdbx_src_id: string;
      /**
       * Reference to the provenance of the source organism details for the entity.
       *  Primary data indicates information obtained from the same source as the structural model.
       *  UniProt and NCBI are provided as alternate sources of provenance for organism details.
       */
      provenance_source?: "NCBI" | "PDB Primary Data" | "Primary Data" | "UniProt";
      /**
       * The scientific name of the source organism assigned by the PDB depositor.
       */
      scientific_name?: string;
      /**
       * The source type for the entity
       */
      source_type?: "genetically engineered" | "natural" | "synthetic";
      taxonomy_lineage?: {
        /**
         * Members of the NCBI Taxonomy lineage as parent taxonomy lineage depth (1-N)
         */
        depth?: number;
        /**
         * Members of the NCBI Taxonomy lineage as parent taxonomy idcodes.
         */
        id?: string;
        /**
         * Memebers of the NCBI Taxonomy lineage as parent taxonomy names.
         */
        name?: string;
      }[];
      rcsb_gene_name?: {
        /**
         * A code indicating the provenance of the source organism details for the entity
         */
        provenance_source?: "NCBI" | "PDB Primary Data" | "Primary Data" | "UniProt";
        /**
         * Gene name.
         */
        value?: string;
      }[];
    }[]
  ];
  rcsb_polymer_entity?: {
    /**
     * A description of special aspects of the entity.
     */
    details?: string;
    /**
     * Formula mass (KDa) of the entity.
     */
    formula_weight?: number;
    /**
     * A description of the polymer entity.
     */
    pdbx_description?: string;
    /**
     * Enzyme Commission (EC) number(s)
     */
    pdbx_ec?: string;
    /**
     * Polymer entity fragment description(s).
     */
    pdbx_fragment?: string;
    /**
     * Details about any polymer entity mutation(s).
     */
    pdbx_mutation?: string;
    /**
     * The number of molecules of the entity in the entry.
     */
    pdbx_number_of_molecules?: number;
    /**
     * A code indicating the entity has multiple biological sources.
     */
    rcsb_multiple_source_flag?: "N" | "Y";
    /**
     * The number of biological sources for the polymer entity. Multiple source contributions
     *  may come from the same organism (taxonomy).
     */
    rcsb_source_part_count?: number;
    /**
     * The number of distinct source taxonomies for the polymer entity. Commonly used to identify chimeric polymers.
     */
    rcsb_source_taxonomy_count?: number;
    /**
     * The method by which the sample for the polymer entity was produced.
     *  Entities isolated directly from natural sources (tissues, soil
     *  samples etc.) are expected to have further information in the
     *  ENTITY_SRC_NAT category. Entities isolated from genetically
     *  manipulated sources are expected to have further information in
     *  the ENTITY_SRC_GEN category.
     */
    src_method?: "man" | "nat" | "syn";
    rcsb_ec_lineage?: {
      /**
       * Members of the enzyme classification lineage as parent classification hierarchy depth (1-N).
       */
      depth?: number;
      /**
       * Members of the enzyme classification lineage as parent classification codes.
       */
      id?: string;
      /**
       * Members of the enzyme classification lineage as parent classification names.
       */
      name?: string;
    }[];
    rcsb_macromolecular_names_combined?: {
      /**
       * Combined list of macromolecular names.
       */
      name?: string;
      /**
       * Combined list of macromolecular names associated provenance code.
       *
       *  ECO (https://github.com/evidenceontology/evidenceontology)
       */
      provenance_code?: string;
      /**
       * Combined list of macromolecular names associated name source.
       */
      provenance_source?: "PDB Preferred Name" | "PDB Synonym";
    }[];
    rcsb_enzyme_class_combined?: {
      /**
       * The enzyme classification hierarchy depth (1-N).
       */
      depth?: number;
      /**
       * Combined list of enzyme class assignments.
       */
      ec?: string;
      /**
       * Combined list of enzyme class associated provenance sources.
       */
      provenance_source?: "PDB Primary Data" | "UniProt";
    }[];
    rcsb_polymer_name_combined?: {
      /**
       * @minItems 1
       */
      names?: [string, ...string[]];
      provenance_source?: "PDB Preferred Name" | "PDB Description" | "UniProt Name";
      [k: string]: unknown;
    };
  };
  /**
   * @minItems 1
   */
  rcsb_polymer_entity_align?: [
    {
      /**
       * Code identifying the individual, organization or program that
       *  assigned the reference sequence.
       */
      provenance_source?: string;
      /**
       * Reference sequence accession code.
       */
      reference_database_accession?: string;
      /**
       * Reference sequence isoform identifier.
       */
      reference_database_isoform?: string;
      /**
       * Reference sequence database name.
       */
      reference_database_name?: "EMBL" | "GenBank" | "NDB" | "NORINE" | "PDB" | "PIR" | "PRF" | "RefSeq" | "UniProt";
      aligned_regions?: {
        /**
         * An identifier for the monomer in the entity sequence at which this segment of the alignment begins.
         */
        entity_beg_seq_id?: number;
        /**
         * An length of the this segment of the alignment.
         */
        length?: number;
        /**
         * An identifier for the monomer in the reference sequence at which this segment of the alignment begins.
         */
        ref_beg_seq_id?: number;
      }[];
    },
    ...{
      /**
       * Code identifying the individual, organization or program that
       *  assigned the reference sequence.
       */
      provenance_source?: string;
      /**
       * Reference sequence accession code.
       */
      reference_database_accession?: string;
      /**
       * Reference sequence isoform identifier.
       */
      reference_database_isoform?: string;
      /**
       * Reference sequence database name.
       */
      reference_database_name?: "EMBL" | "GenBank" | "NDB" | "NORINE" | "PDB" | "PIR" | "PRF" | "RefSeq" | "UniProt";
      aligned_regions?: {
        /**
         * An identifier for the monomer in the entity sequence at which this segment of the alignment begins.
         */
        entity_beg_seq_id?: number;
        /**
         * An length of the this segment of the alignment.
         */
        length?: number;
        /**
         * An identifier for the monomer in the reference sequence at which this segment of the alignment begins.
         */
        ref_beg_seq_id?: number;
      }[];
    }[]
  ];
  /**
   * @minItems 1
   */
  rcsb_polymer_entity_annotation?: [
    {
      /**
       * An identifier for the annotation.
       */
      annotation_id?: string;
      /**
       * Identifies the version of the annotation assignment.
       */
      assignment_version?: string;
      /**
       * A description for the annotation.
       */
      description?: string;
      /**
       * A name for the annotation.
       */
      name?: string;
      /**
       * Code identifying the individual, organization or program that
       *  assigned the annotation.
       */
      provenance_source?: string;
      /**
       * A type or category of the annotation.
       */
      type?: "CARD" | "GO" | "GlyCosmos" | "GlyGen" | "InterPro" | "MemProtMD" | "OPM" | "PDBTM" | "Pfam" | "mpstruc";
      annotation_lineage?: {
        /**
         * Members of the annotation lineage as parent lineage depth (1-N)
         */
        depth?: number;
        /**
         * Members of the annotation lineage as parent class identifiers.
         */
        id?: string;
        /**
         * Members of the annotation lineage as parent class names.
         */
        name?: string;
      }[];
      additional_properties?: {
        /**
         * The additional property name.
         */
        name?: "CARD_ARO_CATEGORY" | "CARD_ARO_CVTERM_ID" | "CARD_ARO_DRUG_CLASS" | "CARD_ARO_RESISTANCE_MECHANISM";
        values?: (string | number | number)[];
      }[];
    },
    ...{
      /**
       * An identifier for the annotation.
       */
      annotation_id?: string;
      /**
       * Identifies the version of the annotation assignment.
       */
      assignment_version?: string;
      /**
       * A description for the annotation.
       */
      description?: string;
      /**
       * A name for the annotation.
       */
      name?: string;
      /**
       * Code identifying the individual, organization or program that
       *  assigned the annotation.
       */
      provenance_source?: string;
      /**
       * A type or category of the annotation.
       */
      type?: "CARD" | "GO" | "GlyCosmos" | "GlyGen" | "InterPro" | "MemProtMD" | "OPM" | "PDBTM" | "Pfam" | "mpstruc";
      annotation_lineage?: {
        /**
         * Members of the annotation lineage as parent lineage depth (1-N)
         */
        depth?: number;
        /**
         * Members of the annotation lineage as parent class identifiers.
         */
        id?: string;
        /**
         * Members of the annotation lineage as parent class names.
         */
        name?: string;
      }[];
      additional_properties?: {
        /**
         * The additional property name.
         */
        name?: "CARD_ARO_CATEGORY" | "CARD_ARO_CVTERM_ID" | "CARD_ARO_DRUG_CLASS" | "CARD_ARO_RESISTANCE_MECHANISM";
        values?: (string | number | number)[];
      }[];
    }[]
  ];
  rcsb_polymer_entity_container_identifiers: {
    asym_ids?: string[];
    auth_asym_ids?: string[];
    chem_comp_monomers?: string[];
    chem_comp_nstd_monomers?: string[];
    /**
     * The chemical reference definition identifier for the entity in this container.
     */
    chem_ref_def_id?: string;
    /**
     * Entity identifier for the container.
     */
    entity_id: string;
    /**
     * Entry identifier for the container.
     */
    entry_id: string;
    /**
     * The BIRD identifier for the entity in this container.
     */
    prd_id?: string;
    /**
     * A unique identifier for each object in this entity container formed by
     *  an underscore separated concatenation of entry and entity identifiers.
     */
    rcsb_id?: string;
    reference_sequence_identifiers?: {
      /**
       * Reference database accession code
       */
      database_accession?: string;
      /**
       * Reference database identifier for the sequence isoform
       */
      database_isoform?: string;
      /**
       * Reference database name
       */
      database_name?: "EMBL" | "GenBank" | "NDB" | "NORINE" | "PDB" | "PIR" | "PRF" | "RefSeq" | "UniProt";
      /**
       * Source of the reference database assignment
       */
      provenance_source?: "PDB" | "RCSB" | "SIFTS" | "UniProt";
    }[];
    /**
     * @minItems 1
     */
    uniprot_ids?: [string, ...string[]];
  };
  /**
   * @minItems 1
   */
  rcsb_polymer_entity_feature?: [
    {
      /**
       * Identifies the version of the feature assignment.
       */
      assignment_version?: string;
      /**
       * A description for the feature.
       */
      description?: string;
      /**
       * An identifier for the feature.
       */
      feature_id?: string;
      /**
       * A name for the feature.
       */
      name?: string;
      /**
       * Code identifying the individual, organization or program that
       *  assigned the feature.
       */
      provenance_source?: string;
      /**
       * Code residue coordinate system for the assigned feature.
       */
      reference_scheme?: "NCBI" | "PDB entity" | "UniProt";
      /**
       * A type or category of the feature.
       */
      type?:
        | "CARD_MODEL"
        | "IMGT_ANTIBODY_DESCRIPTION"
        | "IMGT_ANTIBODY_DOMAIN_NAME"
        | "IMGT_ANTIBODY_GENE_ALLELE_NAME"
        | "IMGT_ANTIBODY_ORGANISM_NAME"
        | "IMGT_ANTIBODY_PROTEIN_NAME"
        | "IMGT_ANTIBODY_RECEPTOR_DESCRIPTION"
        | "IMGT_ANTIBODY_RECEPTOR_TYPE"
        | "Pfam"
        | "SABDAB_ANTIBODY_ANTIGEN_NAME"
        | "SABDAB_ANTIBODY_NAME"
        | "SABDAB_ANTIBODY_TARGET"
        | "artifact"
        | "modified_monomer"
        | "mutation"
        | "hydropathy"
        | "disorder"
        | "disorder_binding";
      feature_positions?: {
        /**
         * An identifier for the leading monomer corresponding to the feature assignment.
         */
        beg_comp_id?: string;
        /**
         * An identifier for the monomer at which this segment of the feature begins.
         */
        beg_seq_id: number;
        /**
         * An identifier for the monomer at which this segment of the feature ends.
         */
        end_seq_id?: number;
        /**
         * The value for the feature over this monomer segment.
         */
        value?: number;
        values?: number[];
      }[];
      additional_properties?: {
        /**
         * The additional property name.
         */
        name?: "CARD_MODEL_DESCRIPTION" | "CARD_MODEL_ORGANISM" | "PARENT_COMP_ID";
        values?: (string | number | number)[];
      }[];
    },
    ...{
      /**
       * Identifies the version of the feature assignment.
       */
      assignment_version?: string;
      /**
       * A description for the feature.
       */
      description?: string;
      /**
       * An identifier for the feature.
       */
      feature_id?: string;
      /**
       * A name for the feature.
       */
      name?: string;
      /**
       * Code identifying the individual, organization or program that
       *  assigned the feature.
       */
      provenance_source?: string;
      /**
       * Code residue coordinate system for the assigned feature.
       */
      reference_scheme?: "NCBI" | "PDB entity" | "UniProt";
      /**
       * A type or category of the feature.
       */
      type?:
        | "CARD_MODEL"
        | "IMGT_ANTIBODY_DESCRIPTION"
        | "IMGT_ANTIBODY_DOMAIN_NAME"
        | "IMGT_ANTIBODY_GENE_ALLELE_NAME"
        | "IMGT_ANTIBODY_ORGANISM_NAME"
        | "IMGT_ANTIBODY_PROTEIN_NAME"
        | "IMGT_ANTIBODY_RECEPTOR_DESCRIPTION"
        | "IMGT_ANTIBODY_RECEPTOR_TYPE"
        | "Pfam"
        | "SABDAB_ANTIBODY_ANTIGEN_NAME"
        | "SABDAB_ANTIBODY_NAME"
        | "SABDAB_ANTIBODY_TARGET"
        | "artifact"
        | "modified_monomer"
        | "mutation"
        | "hydropathy"
        | "disorder"
        | "disorder_binding";
      feature_positions?: {
        /**
         * An identifier for the leading monomer corresponding to the feature assignment.
         */
        beg_comp_id?: string;
        /**
         * An identifier for the monomer at which this segment of the feature begins.
         */
        beg_seq_id: number;
        /**
         * An identifier for the monomer at which this segment of the feature ends.
         */
        end_seq_id?: number;
        /**
         * The value for the feature over this monomer segment.
         */
        value?: number;
        values?: number[];
      }[];
      additional_properties?: {
        /**
         * The additional property name.
         */
        name?: "CARD_MODEL_DESCRIPTION" | "CARD_MODEL_ORGANISM" | "PARENT_COMP_ID";
        values?: (string | number | number)[];
      }[];
    }[]
  ];
  /**
   * @minItems 1
   */
  rcsb_polymer_entity_feature_summary?: [
    {
      /**
       * The feature count.
       */
      count?: number;
      /**
       * The fractional feature coverage relative to the full entity sequence.
       *  For instance, the fraction of features such as mutations, artifacts or modified monomers
       *  relative to the length of the entity sequence.
       */
      coverage?: number;
      /**
       * The maximum feature length.
       */
      maximum_length?: number;
      /**
       * The maximum feature value.
       */
      maximum_value?: number;
      /**
       * The minimum feature length.
       */
      minimum_length?: number;
      /**
       * The minimum feature value.
       */
      minimum_value?: number;
      /**
       * Type or category of the feature.
       */
      type?:
        | "CARD_MODEL"
        | "IMGT_ANTIBODY_DESCRIPTION"
        | "IMGT_ANTIBODY_DOMAIN_NAME"
        | "IMGT_ANTIBODY_GENE_ALLELE_NAME"
        | "IMGT_ANTIBODY_ORGANISM_NAME"
        | "IMGT_ANTIBODY_PROTEIN_NAME"
        | "IMGT_ANTIBODY_RECEPTOR_DESCRIPTION"
        | "IMGT_ANTIBODY_RECEPTOR_TYPE"
        | "Pfam"
        | "SABDAB_ANTIBODY_ANTIGEN_NAME"
        | "SABDAB_ANTIBODY_NAME"
        | "SABDAB_ANTIBODY_TARGET"
        | "artifact"
        | "modified_monomer"
        | "mutation";
    },
    ...{
      /**
       * The feature count.
       */
      count?: number;
      /**
       * The fractional feature coverage relative to the full entity sequence.
       *  For instance, the fraction of features such as mutations, artifacts or modified monomers
       *  relative to the length of the entity sequence.
       */
      coverage?: number;
      /**
       * The maximum feature length.
       */
      maximum_length?: number;
      /**
       * The maximum feature value.
       */
      maximum_value?: number;
      /**
       * The minimum feature length.
       */
      minimum_length?: number;
      /**
       * The minimum feature value.
       */
      minimum_value?: number;
      /**
       * Type or category of the feature.
       */
      type?:
        | "CARD_MODEL"
        | "IMGT_ANTIBODY_DESCRIPTION"
        | "IMGT_ANTIBODY_DOMAIN_NAME"
        | "IMGT_ANTIBODY_GENE_ALLELE_NAME"
        | "IMGT_ANTIBODY_ORGANISM_NAME"
        | "IMGT_ANTIBODY_PROTEIN_NAME"
        | "IMGT_ANTIBODY_RECEPTOR_DESCRIPTION"
        | "IMGT_ANTIBODY_RECEPTOR_TYPE"
        | "Pfam"
        | "SABDAB_ANTIBODY_ANTIGEN_NAME"
        | "SABDAB_ANTIBODY_NAME"
        | "SABDAB_ANTIBODY_TARGET"
        | "artifact"
        | "modified_monomer"
        | "mutation";
    }[]
  ];
  rcsb_polymer_entity_keywords?: {
    /**
     * Keywords describing this polymer entity.
     */
    text?: string;
  };
  /**
   * @minItems 1
   */
  rcsb_polymer_entity_name_com?: [
    {
      /**
       * A common name for the polymer entity.
       */
      name: string;
    },
    ...{
      /**
       * A common name for the polymer entity.
       */
      name: string;
    }[]
  ];
  /**
   * @minItems 1
   */
  rcsb_polymer_entity_name_sys?: [
    {
      /**
       * The systematic name for the polymer entity.
       */
      name: string;
      /**
       * The system used to generate the systematic name of the polymer entity.
       */
      system?: string;
    },
    ...{
      /**
       * The systematic name for the polymer entity.
       */
      name: string;
      /**
       * The system used to generate the systematic name of the polymer entity.
       */
      system?: string;
    }[]
  ];
  /**
   * @minItems 1
   */
  rcsb_related_target_references?: [
    {
      /**
       * The related target data resource name.
       */
      related_resource_name?: "ChEMBL" | "DrugBank" | "Pharos";
      /**
       * The version of the target data resource.
       */
      related_resource_version?: string;
      /**
       * An identifier for the target sequence in the related data resource.
       */
      related_target_id?: string;
      /**
       * NCBI Taxonomy identifier for the target organism.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      target_taxonomy_id?: number;
      aligned_target?: {
        /**
         * The position of the monomer in the entity sequence at which the alignment begins.
         */
        entity_beg_seq_id?: number;
        /**
         * The length of the alignment.
         */
        length?: number;
        /**
         * The position of the monomer in the target sequence at which the alignment begins.
         */
        target_beg_seq_id?: number;
      }[];
    },
    ...{
      /**
       * The related target data resource name.
       */
      related_resource_name?: "ChEMBL" | "DrugBank" | "Pharos";
      /**
       * The version of the target data resource.
       */
      related_resource_version?: string;
      /**
       * An identifier for the target sequence in the related data resource.
       */
      related_target_id?: string;
      /**
       * NCBI Taxonomy identifier for the target organism.
       *
       *  Reference:
       *
       *  Wheeler DL, Chappey C, Lash AE, Leipe DD, Madden TL, Schuler GD,
       *  Tatusova TA, Rapp BA (2000). Database resources of the National
       *  Center for Biotechnology Information. Nucleic Acids Res 2000 Jan
       *  1;28(1):10-4
       *
       *  Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Rapp BA,
       *  Wheeler DL (2000). GenBank. Nucleic Acids Res 2000 Jan 1;28(1):15-18.
       */
      target_taxonomy_id?: number;
      aligned_target?: {
        /**
         * The position of the monomer in the entity sequence at which the alignment begins.
         */
        entity_beg_seq_id?: number;
        /**
         * The length of the alignment.
         */
        length?: number;
        /**
         * The position of the monomer in the target sequence at which the alignment begins.
         */
        target_beg_seq_id?: number;
      }[];
    }[]
  ];
  /**
   * @minItems 1
   */
  rcsb_target_cofactors?: [
    {
      /**
       * The value measured or determined by the assay.
       */
      binding_assay_value?: number;
      /**
       * The type of measurement or value determined by the assay.
       */
      binding_assay_value_type?: "pAC50" | "pEC50" | "pIC50" | "pKd" | "pKi";
      /**
       * Standard IUPAC International Chemical Identifier (InChI) descriptor key
       *  for the cofactor.
       *
       *  InChI, the IUPAC International Chemical Identifier,
       *  by Stephen R Heller, Alan McNaught, Igor Pletnev, Stephen Stein and Dmitrii Tchekhovskoi,
       *  Journal of Cheminformatics, 2015, 7:23
       */
      cofactor_InChIKey?: string;
      /**
       * Simplified molecular-input line-entry system (SMILES) descriptor for the cofactor.
       *
       *    Weininger D (February 1988). "SMILES, a chemical language and information system. 1.
       *    Introduction to methodology and encoding rules". Journal of Chemical Information and Modeling. 28 (1): 31-6.
       *
       *    Weininger D, Weininger A, Weininger JL (May 1989).
       *    "SMILES. 2. Algorithm for generation of unique SMILES notation",
       *    Journal of Chemical Information and Modeling. 29 (2): 97-101.
       */
      cofactor_SMILES?: string;
      /**
       * The chemical component definition identifier for the cofactor.
       */
      cofactor_chem_comp_id?: string;
      /**
       * The cofactor description.
       */
      cofactor_description?: string;
      /**
       * The cofactor name.
       */
      cofactor_name?: string;
      /**
       * The BIRD definition identifier for the cofactor.
       */
      cofactor_prd_id?: string;
      /**
       * Identifier for the cofactor assigned by the resource.
       */
      cofactor_resource_id?: string;
      /**
       * Mechanism of action describes the biochemical interaction through which the
       *  cofactor produces a pharmacological effect.
       */
      mechanism_of_action?: string;
      /**
       * A flag to indicate the cofactor is a structural neighbor of this
       *  entity.
       */
      neighbor_flag?: "N" | "Y";
      patent_nos?: string[];
      pubmed_ids?: number[];
      /**
       * Resource providing target and cofactor data.
       */
      resource_name?: "ChEMBL" | "DrugBank" | "Pharos";
      /**
       * Version of the information distributed by the data resource.
       */
      resource_version?: string;
      /**
       * Identifier for the target assigned by the resource.
       */
      target_resource_id?: string;
    },
    ...{
      /**
       * The value measured or determined by the assay.
       */
      binding_assay_value?: number;
      /**
       * The type of measurement or value determined by the assay.
       */
      binding_assay_value_type?: "pAC50" | "pEC50" | "pIC50" | "pKd" | "pKi";
      /**
       * Standard IUPAC International Chemical Identifier (InChI) descriptor key
       *  for the cofactor.
       *
       *  InChI, the IUPAC International Chemical Identifier,
       *  by Stephen R Heller, Alan McNaught, Igor Pletnev, Stephen Stein and Dmitrii Tchekhovskoi,
       *  Journal of Cheminformatics, 2015, 7:23
       */
      cofactor_InChIKey?: string;
      /**
       * Simplified molecular-input line-entry system (SMILES) descriptor for the cofactor.
       *
       *    Weininger D (February 1988). "SMILES, a chemical language and information system. 1.
       *    Introduction to methodology and encoding rules". Journal of Chemical Information and Modeling. 28 (1): 31-6.
       *
       *    Weininger D, Weininger A, Weininger JL (May 1989).
       *    "SMILES. 2. Algorithm for generation of unique SMILES notation",
       *    Journal of Chemical Information and Modeling. 29 (2): 97-101.
       */
      cofactor_SMILES?: string;
      /**
       * The chemical component definition identifier for the cofactor.
       */
      cofactor_chem_comp_id?: string;
      /**
       * The cofactor description.
       */
      cofactor_description?: string;
      /**
       * The cofactor name.
       */
      cofactor_name?: string;
      /**
       * The BIRD definition identifier for the cofactor.
       */
      cofactor_prd_id?: string;
      /**
       * Identifier for the cofactor assigned by the resource.
       */
      cofactor_resource_id?: string;
      /**
       * Mechanism of action describes the biochemical interaction through which the
       *  cofactor produces a pharmacological effect.
       */
      mechanism_of_action?: string;
      /**
       * A flag to indicate the cofactor is a structural neighbor of this
       *  entity.
       */
      neighbor_flag?: "N" | "Y";
      patent_nos?: string[];
      pubmed_ids?: number[];
      /**
       * Resource providing target and cofactor data.
       */
      resource_name?: "ChEMBL" | "DrugBank" | "Pharos";
      /**
       * Version of the information distributed by the data resource.
       */
      resource_version?: string;
      /**
       * Identifier for the target assigned by the resource.
       */
      target_resource_id?: string;
    }[]
  ];
  /**
   * A unique identifier for each object in this entity container formed by
   *  an underscore separated concatenation of entry and entity identifiers.
   */
  rcsb_id: string;
  /**
   * @minItems 1
   */
  rcsb_polymer_entity_group_membership?: [
    {
      /**
       * A unique identifier for a group of entities
       */
      group_id: string;
      /**
       * Method used to establish group membership
       */
      aggregation_method: "sequence_identity" | "matching_uniprot_accession";
      /**
       * Degree of similarity expressed as a floating-point number
       */
      similarity_cutoff?: number;
      aligned_regions?: {
        /**
         * An identifier for the monomer in the entity sequence at which this segment of the alignment begins.
         */
        entity_beg_seq_id?: number;
        /**
         * An length of the this segment of the alignment.
         */
        length?: number;
        /**
         * An identifier for the monomer in the reference sequence at which this segment of the alignment begins.
         */
        ref_beg_seq_id?: number;
      }[];
    },
    ...{
      /**
       * A unique identifier for a group of entities
       */
      group_id: string;
      /**
       * Method used to establish group membership
       */
      aggregation_method: "sequence_identity" | "matching_uniprot_accession";
      /**
       * Degree of similarity expressed as a floating-point number
       */
      similarity_cutoff?: number;
      aligned_regions?: {
        /**
         * An identifier for the monomer in the entity sequence at which this segment of the alignment begins.
         */
        entity_beg_seq_id?: number;
        /**
         * An length of the this segment of the alignment.
         */
        length?: number;
        /**
         * An identifier for the monomer in the reference sequence at which this segment of the alignment begins.
         */
        ref_beg_seq_id?: number;
      }[];
    }[]
  ];
  /**
   * @minItems 1
   */
  rcsb_genomic_lineage?: [
    {
      /**
       * Automatically assigned ID that uniquely identifies taxonomy, chromosome or gene in the Genome Location Browser.
       */
      id?: string;
      /**
       * A human-readable term name.
       */
      name?: string;
      /**
       * Classification hierarchy depth.
       */
      depth?: number;
      [k: string]: unknown;
    },
    ...{
      /**
       * Automatically assigned ID that uniquely identifies taxonomy, chromosome or gene in the Genome Location Browser.
       */
      id?: string;
      /**
       * A human-readable term name.
       */
      name?: string;
      /**
       * Classification hierarchy depth.
       */
      depth?: number;
      [k: string]: unknown;
    }[]
  ];
  /**
   * Information about integral membrane proteins whose crystallographic, or sometimes NMR or cryo-EM, structures have been determined to a resolution sufficient to identify TM helices of helix-bundle membrane proteins (typically 4 - 4.5 A). Mpstruc provides manually curated information about integral membrane proteins. These manual annotations are extended using sequence clusters and according to the following procedure:Single chain transmembrane proteins: Any PDB chain sharing 90% sequence identity to this transmembrane protein is assigned as a transmembrane protein as well, and shares the same transmembrane annotation.Multi-chain transmembrane proteins:If the reference mpstruc entry contains multiple protein entities, it is necessary to identify which of the entities are presumed to be transmembrane chains. This is done in conjunction with Uniprot annotations. Transmembrane protein entities are identified by checking if their corresponding Uniprot sequence has annotations labeled <i>transmembrane</i> or <i>intramembrane region</i>. For transmembrane entities, all members of the sequence cluster (90% sequence identity) are programmatically infered to be members of the same class of transmembrane proteins by applying the above procedure for single entity mpstruc entries. Annotations are denoned with 'Homology' provenance code.
   *
   * @minItems 1
   */
  rcsb_membrane_lineage?: [
    {
      /**
       * Automatically assigned ID for membrane classification term in the Membrane Protein Browser.
       */
      id?: string;
      /**
       * Membrane protein classification term.
       */
      name?: string;
      /**
       * Hierarchy depth.
       */
      depth?: number;
    },
    ...{
      /**
       * Automatically assigned ID for membrane classification term in the Membrane Protein Browser.
       */
      id?: string;
      /**
       * Membrane protein classification term.
       */
      name?: string;
      /**
       * Hierarchy depth.
       */
      depth?: number;
    }[]
  ];
  /**
   * Mpstruc keyword denotes original annotation, Homology keyword denotes annotation inferred by homology.
   */
  rcsb_membrane_lineage_provenance_code?: "Mpstruc" | "Homology";
  /**
   * Indicates intrinsic flexibility of protein structures determined from structural variations between different depositions and chains in asymmetric units of the same protein in PDB (95% sequence identity).
   */
  rcsb_cluster_flexibility?: {
    /**
     * Link to the associated PDBFlex database entry.
     */
    link?: string;
    /**
     * Structural flexibility in the cluster (95% sequence identity) where a given entity belongs.
     */
    label?: string;
    /**
     * Average RMSD refer to average pairwise RMSD (Root Mean Square Deviation of C-alpha atoms) between structures in the cluster (95% sequence identity) where a given entity belongs.
     */
    avg_rmsd?: number;
    /**
     * Maximal RMSD refer to maximal pairwise RMSD (Root Mean Square Deviation of C-alpha atoms) between structures in the cluster (95% sequence identity) where a given entity belongs.
     */
    max_rmsd?: number;
    provenance_code?: "PDBFlex";
  };
  rcsb_latest_revision?: {
    /**
     * The major version number of the latest revision.
     */
    major_revision?: number;
    /**
     * The minor version number of the latest revision.
     */
    minor_revision?: number;
    /**
     * The release date of the latest revision item.
     */
    revision_date?: string;
    [k: string]: unknown;
  };
}
